core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java

```diff
@@ -462,44 +454,50 @@ private void connect(
     }
 
     private void onSuccessfulReconnect() {
-      // If reconnectOnFailure was true and we've never connected before, complete the future now,
-      // otherwise it's already complete and this is a no-op.
-      initFuture.complete(null);
+      // If reconnectOnFailure was true and we've never connected before, complete the future now to
+      // signal that the initialization is complete.
+      boolean isFirstConnection = initFuture.complete(null);
```

Comment:
This method is only called from the lambda in `reconnect()`, which in turn is only called from the method reference created when the `Reconnection` object is created. So can this `isFirstConnection` variable ever be true, and if so, in which case?

Reply:
Yes, if the reconnection is not part of the initial attempt:
- the driver connects successfully.
- later, the control node goes down.
- `onChannelClosed` is invoked, it calls `reconnection.start()`
- the `Reconnection` object starts scheduling attempts. Each attempt will invoke `reconnect()`. If the attempt fails, `Reconnection` keeps trying. Otherwise, `onSuccessfulReconnect` is invoked.